src/back/TicketManager/src/Core/TicketManager.Core.Domain/Events/Event.cs

```diff
@@ -23,5 +24,66 @@ public Event(Guid organizerId, string name, string description, string location,
         Description = description;
         Location = location;
         Date = date;
+        Status = EventStatus.Verified;
+    }
+
+    public void ChangeEventStatus(EventStatus status)
+    {
+        switch(status)
+        {
+            case EventStatus.Verified:
+                if(Status != EventStatus.Unverified &&  Status != EventStatus.Cancelled)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Verified from current status.");
+                }
+                break;
+            case EventStatus.Published:
+                if(Status != EventStatus.Verified)
```

Comment:
Shouldn't event be verified to publish it in the first place?

Reply:
Yeah, so if it is not `Verified` we throw an exception